lib/mongo/server/connection.rb

```diff
@@ -167,6 +167,7 @@ def ping
       def timeout
         @timeout ||= options[:socket_timeout]
       end
+      alias :socket_timeout :timeout
```

Comment:
I've marked ``timeout`` as deprecated in Server::Monitor::Connection and Server::Connection to favor the more explicit ``socket_timeout`` method name.